Core/Tornado/Server/TornadoService.py

```diff
@@ -0,0 +1,526 @@
+  # and we should not write back to the client in a different thread.
+  # See https://www.tornadoweb.org/en/branch5.1/web.html#thread-safety-notes
+  # def export_streamToClient(self, filename):
+  #   # https://bhch.github.io/posts/2017/12/serving-large-files-with-tornado-safely-without-blocking/
+  #   #import ipdb; ipdb.set_trace()
+  #   # chunk size to read
+  #   chunk_size = 1024 * 1024 * 1  # 1 MiB
+
+  #   with open(filename, 'rb') as f:
+  #     while True:
+  #       chunk = f.read(chunk_size)
+  #       if not chunk:
+  #         break
+  #       try:
+  #         self.write(chunk)  # write the chunk to response
+  #         self.flush()  # send the chunk to client
+  #       except StreamClosedError:
+  #         # this means the client has closed the connection
+  #         # so break the loop
+  #         break
+  #       finally:
+  #         # deleting the chunk is very important because
+  #         # if many clients are downloading files at the
+  #         # same time, the chunks in memory will keep
+  #         # increasing and will eat up the RAM
+  #         del chunk
+  #         # pause the coroutine so other handlers can run
+  #         yield gen.sleep(0.000000001)  # 1 nanosecond
+
+  #   return S_OK()
+
+  @gen.coroutine
+  def __executeMethod(self):
+    """
+      Execute the method called, this method is executed in an executor
+      We have several try except to catch the different problem who can occurs
+
+      - First, the method does not exist => Attribute error, return an error to client
+      - second, anything happend during execution => General Exception, send error to client
+    """
+
+    # getting method
+    try:
+      method = getattr(self, 'export_%s' % self.method)
+    except AttributeError as e:
+      self._httpError = HTTPErrorCodes.HTTP_NOT_IMPLEMENTED
+      return S_ERROR("Unknown method %s" % self.method)
+
+    # Decode args
+    args_encoded = self.get_body_argument('args', default=encode([]))
+
+    args = decode(args_encoded)[0]
+    # Execute
+    try:
+      self.initializeRequest()
+      retVal = method(*args)
+    except Exception as e:  # pylint: disable=broad-except
+      gLogger.exception("Exception serving request", "%s:%s" % (str(e), repr(e)))
+      retVal = S_ERROR(repr(e))
+      self._httpError = HTTPErrorCodes.HTTP_INTERNAL_SERVER_ERROR
+
+    return retVal
+
+  def __write_return(self, dictionnary):
+    """
+      Write to client what we wan't to return to client
+      It must be a dictionnary
+    """
+
+    # In case of error in server side we hide server CallStack to client
+    if 'CallStack' in dictionnary:
+      del dictionnary['CallStack']
+
+    # Write status code before writing, by default error code is "200 OK"
+    self.set_status(self._httpError)
+
+    # See 4.5.1 http://www.rfc-editor.org/rfc/rfc2046.txt
+    self.set_header("Content-Type", "application/octet-stream")
```

Comment:
Maybe here is where `Content-Encoding: gzip` should be used?

Reply:
Probably ! I'll have to check how to do that properly :-)

Reply:
So it is done automatically when using the `compress_response` argument of the Tornado `Application`